### PR TITLE
feat: Add investigations app backend

### PR DESCRIPTION
**Pull request by fcjack**
**What is this feature?**

This is the backend app that will support the investigations plugin.

**Why do we need this feature?**

The new investigations-app is a plugin frontend only for now, that needs a storage layer that will enable future improvements, like sharing the investigation.

**Who is this feature for?**

For any user that has the investigations plugin enabled.

**Which issue(s) does this PR fix?**:

https://github.com/grafana/investigations-app/issues/12

**Special notes for your reviewer:**

Please check that:
- [X] It works as expected from a user's perspective.
- [X] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: apps/investigation/kinds/manifest.cue [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package kinds
+
+manifest: {
+	// appName is the unique name of your app. It is used to reference the app from other config objects,
+	// and to generate the group used by your app in the app platform API.
+	appName: "investigations-app"
```
Comment (radiohead) [inline apps/investigation/kinds/manifest.cue:6]: I think `-app` is redundant here:
```suggestion
	appName: "investigations"
```

File: apps/investigation/kinds/investigation.cue [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	scope: "Namespaced"
+	codegen: {
+		frontend: false
+		backend:  true
+	}
+	pluralName: "Investigations"
+	current: "v1-alpha"
+	versions: {
+		"v1-alpha": {
+			version: "v1-alpha"
```
Comment (radiohead) [inline apps/investigation/kinds/investigation.cue]: Versions should be alphanumeric. Also, you don't have to specify the version name, it's inferred from the key name:

```suggestion
	current: "v1alpha1"
	versions: {
		"v1alpha1": {
```

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -68,6 +68,7 @@
 
 /apps/alerting/ @grafana/alerting-backend @grafana/alerting-frontend
 /apps/playlist/ @grafana/grafana-app-platform-squad
+/apps/investigation/ @grafana/grafana-app-platform-squad
```
Comment (radiohead) [inline .github/CODEOWNERS:71]: @fcjack I've put this is as a placeholder – feel free to change this to the correct team!

Comment (fcjack) [reply .github/CODEOWNERS]: For now will be me and @matryer , until be ready for the team to assume.

File: pkg/registry/apps/apps.go [hunk 4]
```diff
@@ -40,11 +44,19 @@ func ProvideRegistryServiceSink(
 		RestConfigGetter: cfgWrapper,
 		APIRegistrar:     registrar,
 	}
-	runner, err := runner.NewAPIGroupRunner(cfg, playlistAppProvider)
+
+	var apiGroupRunner *runner.APIGroupRunner
+	var err error
+	if features.IsEnabledGlobally(featuremgmt.FlagInvestigationsBackend) {
```
Comment (radiohead) [inline pkg/registry/apps/apps.go:50]: @toddtreece FYI, I think we should look into having a generic FF here for enabling various APIs, similar to what `kube-apiserver` has in `--runtime-config`.

**Review by radiohead — APPROVED** [review]

File: pkg/registry/apps/apps.go (hunk 4 above)
Comment (toddtreece) [reply pkg/registry/apps/apps.go:50]: @radiohead i think the main issue with a generic toggle is that feature toggles are booleans, so i'm not sure how we could do that unless we enable a group of them with a single toggle

we could use apienablement's `--runtime-config` for this purpose, and we probably should move to that in code and have the existing feature toggles add the API to `runtime-config`. the primary issue with using `runtime-config` directly in HG is that feature toggles have additional plumbing that allow us to control the roll out of new features, so we'd need to add something similar for APIs enabled via `--runtime-config` if we expose that directly via the grafana config

**Pull request merged**